src/file/FileDialog.tsx

```diff
@@ -215,6 +218,74 @@ const OpenLocalFile: React.FC<SourceTabProps> = ({ onDismiss }) => {
     );
 };
 
+const ImportFromString: React.FC<SourceTabProps> = ({ onDismiss }) => {
+    const loadScene = useLoadScene();
+    const setSavedState = useSetSavedState();
+    const isDirty = useIsDirty();
+    const theme = useTheme();
+    const [data, setData] = useState<string | undefined>('');
+    const [error, setError] = useState<string | undefined>('');
+
+    const importCallback = useCallback(async () => {
+        if (!data) {
+            return;
+        }
+
+        if (isDirty) {
+            if (!(await confirmUnsavedChanges(theme))) {
+                return;
+            }
+        }
+
+        let scene;
+        try {
+            scene = await textToScene(data);
+        } catch (ex) {
+            console.log(ex);
+            setError('Invalid Import String');
+            return;
+        }
+
+        loadScene(scene, undefined);
+        setSavedState(scene);
+        onDismiss?.();
+    }, [data, isDirty, theme, loadScene, setSavedState, onDismiss]);
+
+    const onKeyChange = useCallback(
+        (ev: FormEvent<HTMLInputElement | HTMLTextAreaElement>, value?: string) => {
+            setData(value);
+            setError(undefined);
+        },
+        [setError, setData],
+    );
+
+    const onKeyPress = useCallback(
+        (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+            if (ev.key === 'Enter') {
+                importCallback();
+            }
+        },
+        [importCallback],
+    );
+
+    return (
+        <>
+            <TextField
+                label="Enter Import String"
```

Comment:
Kind of pedantic, but this is asking for a thing called an "import string", but there's nothing else in the UI called that, so it might be a little unclear what this means. I can't come up with anything that I think is clearly the best text to use here, but what do you think about something more like "Paste shared string here" to make it clear that the string should come from the share button (and you probably want to copy/paste it)?

Reply:
I agree this could be confusing. The word "here" feels redundant so ill drop it, but I'll go with your suggestion with my next change.

Another option would be to give these strings names, such as "Plan Code". The share tab could then be called "Plan Code", and the tab in the Open menu could also be "Plan Code", with the instruction "Enter Plan Code". If you like this idea I can make those changes as well.

Reply:
I like the "plan code" idea. That makes it clear that it's a very specific type of text.